code/modules/antagonists/wizard/abilities/knock.dm

```diff
@@ -23,17 +23,25 @@
 			SPrange = 5
 		else
 			boutput(holder.owner, SPAN_ALERT("Your spell only works at point blank without a staff to focus it!"))
+
+		var/obj/storage/secure/locked = src.holder.owner.loc
+		if (locked && istype(locked))
+			locked.locked = FALSE
+			locked.open()
+			if (!locked.welded)
+				src.holder.owner.set_loc(locked.loc)
+
 		for(var/obj/machinery/door/G in oview(SPrange, holder.owner))
 			SPAWN(1 DECI SECOND)
 				G.open()
 		for(var/obj/storage/F in oview(SPrange, holder.owner))
```

Comment:
```suggestion
		for(var/obj/storage/F in view(SPrange, holder.owner))
```
Just this should make it affect the locker you're in

Reply:
I tried that just now and it seems to not work with view. Not sure why since view should include the centre.